### DISCUSSION TITLE
Is postal supporting clustering? If yes, is there any documentation available in this regard?

### DISCUSSION THREAD

**Discussion by ghevge** [postalserver/postal#2571] [Help with using Postal]
Is postal supporting clustering? If yes, is there any documentation available in this regard?

I am trying to install multiple postal instances accross multiple phisical servers,  that would like to behave like a single one.

Thanks

**Comment by willpower232** [comment]
There isn't any documentation but as long as you are pointing all processes of Postal to the same RabbitMQ and MySQL, then all you need to do is make sure you have the IP addresses and DNS records in order and you can host as many copies of as many bits of Postal as you like.

**Comment by ghevge** [comment]
Hmm... I was envisioning a HA cluster setup: independent Postal stacks (postal + rabbit + MySQL) on each node. Horizontal scaling.

**Reply by willpower232** [reply]
As long as the RabbitMQs and MySQLs were in some form of cluster and sharing their data equally between them all, that would probably be fine.

I would be slightly concerned about the RabbitMQs having a message consumed simultaneously by multiple instances of Postal and the end user receving two emails if there wasn't say a single URL for accessing the cluster of RabbitMQs but I don't really know much about how they cluster.
